app-runner/Private/DeviceProviders/SauceLabsProvider.ps1

```diff
@@ -587,8 +612,176 @@ class SauceLabsProvider : DeviceProvider {
+
+    <#
+    .SYNOPSIS
+    Handles errors from CopyDeviceItem with helpful diagnostic information.
+    #>
+    [void] HandleCopyDeviceItemError([System.Management.Automation.ErrorRecord]$Error, [string]$DevicePath) {
+        $errorMsg = "Failed to copy file from device: $DevicePath. Error: $($Error.Exception.Message)"
+
+        # Add platform-specific troubleshooting for server errors
+        if ($Error.Exception.Message -match "500|Internal Server Error") {
+            $errorMsg += "`n`nTroubleshooting $($this.MobilePlatform) file access:"
+            $errorMsg += "`n- App Package/Bundle ID: '$($this.CurrentPackageName)'"
+            $errorMsg += "`n- Requested path: '$DevicePath'"
+
+            if ($this.MobilePlatform -eq 'iOS') {
+                try {
+                    $appInfo = $this.CheckAppFileSharingCapability()
+                    if ($appInfo.AllApps -and $appInfo.AllApps.Count -gt 0) {
+                        $errorMsg += "`n- Available apps: $($appInfo.AllApps -join ', ')"
+                        if ($appInfo.Found -and -not $appInfo.FileSharingEnabled) {
+                            $errorMsg += "`n- App found but UIFileSharingEnabled=false"
+                        }
+                    }
+                } catch {
+                    $errorMsg += "`n- Could not check app capabilities: $($_.Exception.Message)"
+                }
+
+                $errorMsg += "`n`nCommon iOS causes:"
+                $errorMsg += "`n1. App missing UIFileSharingEnabled=true in info.plist"
+                $errorMsg += "`n2. File doesn't exist on device"
+                $errorMsg += "`n3. Incorrect path format - must use @bundle.id:documents/relative_path"
+
+                if ($this.CurrentPackageName) {
+                    $errorMsg += "`n`nRequired iOS format: @$($this.CurrentPackageName):documents/relative_path"
+                }
+            }
+            elseif ($this.MobilePlatform -eq 'Android') {
+                $errorMsg += "`n`nMost likely cause: App not built with debuggable flag"
+                $errorMsg += "`n"
+                $errorMsg += "`nFor Android internal storage access (/data/data/...), the app MUST be built with:"
+                $errorMsg += "`n  android:debuggable='true' in AndroidManifest.xml"
+                $errorMsg += "`n"
+                $errorMsg += "`nOther possible causes:"
+                $errorMsg += "`n2. File doesn't exist on device (less likely)"
+                $errorMsg += "`n3. Incorrect path format or permissions"
+
+                if ($this.CurrentPackageName) {
+                    $errorMsg += "`n`nWorking path formats:"
+                    $errorMsg += "`n- Internal storage: /data/data/$($this.CurrentPackageName)/files/app.log (needs debuggable=true)"
+                    $errorMsg += "`n- App-relative: @$($this.CurrentPackageName)/files/app.log (needs debuggable=true)"
+                }
+            }
+        }
+
+        Write-Error $errorMsg
+        throw
```

Comment:
Write-error doesn't just log. with our ErrorActionPreference, it's functionally equivalent to throwing. so the next line is never reached. You can use Write-Warning instead